File: app/src/main/java/dev/mike/couchtour/CuratedMatches.kt
package dev.mike.couchtour

import android.content.Context
import android.util.Log
import kotlinx.serialization.json.Json

object CuratedMatches {
    private var mappings: Map<String, ExternalRelease> = emptyMap()

    fun init(context: Context) {
        runCatching {
            val jsonStr = context.assets.open("curated_releases.json").bufferedReader().use { it.readText() }
            val format = Json { ignoreUnknownKeys = true }
            mappings = format.decodeFromString(jsonStr)
        }.onFailure { e ->
            Log.e("CuratedMatches", "Failed to load curated releases", e)
        }
    }

    fun match(backend: Backend, artistId: String, date: String): ExternalRelease? {
        return mappings["${backend.id}:$artistId:$date"]
    }
}

File: macos/Packages/CouchTourKit/Sources/CouchTourKit/CuratedMatches.swift
import Foundation

public struct CuratedMatches: Sendable {
    public static let shared = CuratedMatches()
    
    private let mappings: [String: ExternalRelease]
    
    private init() {
        if let url = Bundle.module.url(forResource: "curated_releases", withExtension: "json"),
           let data = try? Data(contentsOf: url),
           let decoded = try? JSONDecoder().decode([String: ExternalRelease].self, from: data) {
            self.mappings = decoded
        } else {
            self.mappings = [:]
        }
    }
    
    public func match(backend: Backend, artistId: String, date: String) -> ExternalRelease? {
        let key = "\(backend.rawValue):\(artistId):\(date)"
        return mappings[key]
    }
}
